src/devices/bus/snes/event.cpp

```diff
@@ -152,7 +155,8 @@ void sns_pfest94_device::chip_write(offs_t offset, uint8_t data)
 	{
 		// DSP access
 		offset &= 0x1fff;
-		m_upd7725->snesdsp_write(offset < 0x1000, data);
+		if (BIT(~offset, 12))
+			m_upd7725->data_w(data);
```

Comment:
Will bit 12 being clear actually prevent the write from happening, or will the chip still be selected and treat it as a data write in both cases?

Also, you can get rid of the `offset &= 0x1fff` in these functions since you’re only testing one bit anyway.

Reply:
Offset bit 12 is data/status mode select at both read/write accessing, and status write is ignored at write case.